Sources/JobsRedisDriver/JobsRedisDriver.swift

```diff
@@ -4,38 +4,40 @@ import NIO
 import Foundation
 import Vapor
 
-/// A wrapper that conforms to `JobsPersistenceLayer`
+/// A wrapper that conforms to `JobsDriver`
 public struct JobsRedisDriver: JobsDriver {
     /// The `RedisClient` to run commands on
     public let client: RedisClient
-
-    let logger: Logger
     
-    /// The `EventLoop` to run jobs on
-    public var eventLoop: EventLoop {
-        return self.client.eventLoop
-    }
+    /// The event loop group
+    public let eventLoopGroup: EventLoopGroup
+    
+    /// The logger
+    let logger: Logger
     
     /// Creates a new `RedisJobs` instance
     ///
     /// - Parameters:
     ///   - client: The `RedisClient` to use
-    public init(client: RedisClient) {
+    public init(client: RedisClient, eventLoopGroup: EventLoopGroup) {
```

Comment:
@mcdappdev  We should ensure worker's EventLoop equal to redisConnect's EventLoop.

Maybe below changes are better.

Change `init(client: RedisClient, eventLoopGroup: EventLoopGroup)` to `init(client: ConnectionPool<RedisConnectionSource>)`
Change `func get(key: String, eventLoop: JobsEventLoopPreference)` to `func get(key: String, eventLoop: EventLoop)`
And change `self.client.rpoplpush` to `self.client.withConnection(eventLoop: .delegate(on: eventLoop)) { redis in redis.rpoplpush }`

Reply:
That would involve changing the Jobs protocol which is solid currently, I think there needs to be another solution. @tanner0101 Thoughts?